### ISSUE TITLE
getObjects fails on specific index because of null-safety

### ISSUE THREAD

**Issue by JustinHorn** [knoxpo/dart_algolia#51]
Dart version: [eg: 2.12.0]
Flutter version: [eg: 2.0.0 (stable)]

Creating a query and then calling getObjects() would throw a null-safety exception with one of two algolia apps.

This is due to the index returning less data than expected for example:
"nbHits" was missing, but this package was expecting it and therefore throwing an error.

Another index returned the missing params with the same query. So the different responses are probably caused by the configurations of the algolia apps in the backend. Sadly I can not check why the API responses were different, because I do not have access to the two apps anymore.

**Comment by JustinHorn** [comment]
I workedarround this  2 weeks ago by changing the class `AlgoliaQuerySnapshot` to:
```
part of algolia;

class AlgoliaQuerySnapshot {
  AlgoliaQuerySnapshot._(algolia, index, Map<String, dynamic> map)
      : algolia = algolia,
        index = index,
        _map = map,
        hits = (map['hits'] as List<dynamic>? ?? [])
            .map((dynamic o) {
              var newMap = Map<String, dynamic>.from(o);
              return AlgoliaObjectSnapshot._(algolia, index, newMap);
            })
            .toList()
            .cast<AlgoliaObjectSnapshot>(),
        empty = (map['hits'] as List<dynamic>? ?? []).isEmpty,
        nbHits = map['nbHits'] ?? -1,
        page = map['page'] ?? -1,
        nbPages = map['nbPages'] ?? -1,
        hitsPerPage = map['hitsPerPage'] ?? -1,
        processingTimeMS = map['processingTimeMS'] ?? -1,
        exhaustiveNbHits = map['exhaustiveNbHits'],
        query = map['query'] ?? '',
        params = map['params'] ?? '',
        facets = map['facets'] ?? {},
        facetsStats = map['facets_stats'] ?? {};

  final List<AlgoliaObjectSnapshot> hits;
  final Algolia algolia;
  final bool empty;
  final int nbHits;
  final int page;
  final int nbPages;
  final int hitsPerPage;
  final int processingTimeMS;
  final bool? exhaustiveNbHits;
  final String query;
  final String params;
  final String index;
  final Map<String, dynamic> facets;
  final Map<String, dynamic> facetsStats;
  final Map<String, dynamic> _map;

  @override
  String toString() {
    return _map.toString();
  }

  Map<String, dynamic> toMap() {
    return _map;
  }

  bool get hasHits {
    return hits.isNotEmpty;
  }
}
```
I did not test whether these changes break certain parts of the package, but creating a query and calling getObjects() worked.

**Comment by idrish-sorathiya** [comment]
Receiving: 
_TypeError (type 'Null' is not a subtype of type 'int')

This is coming at : `page = map['page'],`

Hey, this is blocking me. Please let me know if there is any quick solution for it.

**Comment by zixuan-espressoft** [comment]
I have this issue too, hope it can be fixed.
![Screenshot 2021-07-04 at 3 39 09 PM](https://user-images.githubusercontent.com/41556990/124377333-09312c00-dcde-11eb-8894-21c90c5e6893.png)

**Comment by nhathiwala** [comment]
This issue have been fixed in version `^1.0.2`
